root/usr/share/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -106,22 +106,29 @@ def perform_upload proxy_url, proxy_type, custom_facts
   error_box("Unable to upload facts", e)
 end
 
-def new_custom_facts mac
-  ip_cidr, gw, dns = detect_ipv4_credentials('primary')
-  ip = ip_cidr.split('/').first
-  mask = IPAddr.new(ip_cidr).inspect.scan(/\d+.\d+.\d+.\d+\/\d+.\d+.\d+.\d+/).first.split('/').last
-  {
-    'discovery_kexec' => command("kexec --version"),
-    'discovery_ip_cidr' => ip_cidr,
-    'discovery_ip' => ip,
-    'discovery_netmask' => mask,
-    'discovery_gateway' => gw,
-    'discovery_dns' => dns,
-    'discovery_bootif' => mac,
-  }
-rescue => e
+def new_custom_facts(mac)
+  custom_facts = {}
+  custom_facts['discovery_bootif'] = mac
+  custom_facts['discovery_kexec'] = command('kexec --version')
+
+  ip_cidr4, gw4, dns4 = detect_ipv4_credentials('primary')
+  ip_cidr6, gw6, dns6 = detect_ipv6_credentials('primary')
+
+  ip4 = ip_cidr4.split('/').first if ip_cidr4
+  ip6 = ip_cidr6.split('/').first if ip_cidr6
```

Comment:
One thing is that this does not ignore local addresses like `fe80::2aa:bbff:fe17:1701`.

So if the host has IPv4 and local IPv6, the IPv6 will be prioritized in the facts. I don't think it will affect the provisioning process; it is just something to consider.

Reply:
Maybe we should filter out local prefixes like `fe80`?